### PR TITLE
Document PatchNANs input type (issue #18000)

### PR DESCRIPTION
**Pull request by gsmatos**
Fixes https://github.com/opencv/opencv/issues/18000

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/core/include/opencv2/core.hpp [hunk 1]
```diff
@@ -1616,6 +1616,7 @@ CV_EXPORTS_W bool checkRange(InputArray a, bool quiet = true, CV_OUT Point* pos
                             double minVal = -DBL_MAX, double maxVal = DBL_MAX);
 
 /** @brief converts NaN's to the given number
+@param a input/output matrix (CV_32FC1 type).
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core.hpp:1619]: I propose to replace `CV_32FC1` with `CV_32F`, amount of channels does not matter.

**Comment by asmorkalov** [comment]
Thanks for the contribution. This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no need to re-open PR, apply changes "inplace".

**Comment by asmorkalov** [comment]
Doxygen documentation with CI bot reports new warnings. You need to document all function parameters to fix the CI warning.

File: modules/core/include/opencv2/core.hpp (hunk 1 above)
Comment (gsmatos) [reply modules/core/include/opencv2/core.hpp]: Ok, fixed

**Pull request merged**